example/e2e/deploy-federated-nginx.sh

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script has an end to end steps in deploying a federated deployment
+
+set -o nounset
+set -o pipefail
+
+function wait-for-conditions() {
```

Comment:
This should be imported from the scripts dir rather than redefined.

Reply:
I implement the customized version:
1. I try to check 2 conditions for different clusters in parallel. Not one by one.
2. example script refer to development tool script seems not good.

Reply:
The script is _developer only_.  It's intended to validate the instructions of the walkthrough in ci, and is not intended to be run by a user.  Checking conditions in parallel is not required for this use case.

Reply:
when it becomes a developer only scirpt? I'm confused.

https://github.com/kubernetes-sigs/kubefed/issues/104
> There should be a relatively simple walkthrough that guides users through the experience of kicking the tires on federation-v2 by deploying a very simple application. Other attributes of this:

Reply:
The point of #104 was originally focused on ensuring that the example walkthrough works by writing a script.  The script wasn't intended to replace the walkthrough.  It was to exist in addition to the walkthrough to automatically validate in CI that the steps described in the walkthrough would work.  My modification to #104 was to ask for the walkthrough to be updated to showcase more kubefed features (like `kubefedctl federate` and propagation status).

So the deliverables for #104 are:

 - an updated walkthrough in the userguide that demonstrates more kubefed features
 - a script that executes the commands documented in the walkthrough and verifies that those steps work as expected